src/SaveHandler/MongoDB.php

```diff
@@ -57,8 +57,10 @@ public function __construct($mongo, MongoDBOptions $options)
     {
         if (!($mongo instanceof \MongoClient || $mongo instanceof \Mongo)) {
             throw new InvalidArgumentException(
-                'Parameter of type %s is invalid; must be MongoClient or Mongo',
-                (is_object($mongo) ? get_class($mongo) : gettype($mongo))
+                sprintf(
+                    'Parameter of type %s is invalid; must be MongoClient or Mongo',
+                    (is_object($mongo) ? get_class($mongo) : gettype($mongo))
+                )
```

Comment:
The error is caused by the fact the original PR was merged without proper tests: https://github.com/zendframework/zf2/pull/3221 If there was a unittest for this exception the problem would have been detected immediately, that is why it is important to add unittests. BTW a unittest is missing for this PR too :wink: